### ISSUE TITLE
Wrong document executer registered for subscription support

### ISSUE THREAD

**Issue by Shane32** [graphql-dotnet/server#588]
Due to the following code, `DocumentExecuter` is registered instead of `SubscriptionDocumentExecuter`, and there is nothing to override this within `.AddWebSockets()`.

https://github.com/graphql-dotnet/server/blob/4bb8cc3e87a92c179f2af8b2422fe303d51ef0c5/src/Core/Extensions/ServiceCollectionExtensions.cs#L53

This causes the following error to be thrown when attempting to use subscriptions, unless a specific implementation of `IDocumentExecuter` has been registered within the DI framework:

```
DocumentExecuter does not support executing subscriptions. You can use SubscriptionDocumentExecuter from GraphQL.Subscriptions package to execute subscriptions.
```

Current workaround is to register the correct document executer manually:

```cs
services.AddSingleton<IDocumentExecuter, SubscriptionDocumentExecuter>();
```

This is either a bug, or not properly documented and demonstrated in the readme.

See https://github.com/graphql-dotnet/graphql-dotnet/pull/1973

**Comment by Presipope** [comment]
Ah yes! I've been searching for where to implement the reader in the ReadMe and I couldn't figure out. Thanks for the workaround

**Comment by Shane32** [comment]
Closed by #653
